Examples/LocalDebugging/MyApp/MyApp/ContentView.swift

```diff
@@ -58,22 +67,16 @@ struct ContentView: View {
             let (data, response) = try await URLSession.shared.data(for: request)
 
             guard let httpResponse = response as? HTTPURLResponse else {
-                throw CommunicationError(reason: "invalid response, expected HTTPURLResponse")
+                return "invalid response, expected HTTPURLResponse"
```

Comment:
Uh yes.

I changed it into `() async -> String` because we need to ensure the label is changed in every path.  `throws` could be preserved here and it doesn't affect.

Reply:
I would prefer something like this:
```swift
func register() await -> String {
    do {
        return self.runNetworkRegister(name: self.name, password: self.password)
    } catch as CommunicationError {
        return error.reason
    } catch {
        return error.localizedDescription
    }
}
```

Reply:
I would rather suggest we `catch` in the button action? The `register() [async] [throws]-> String` API is expected not to confuse an actual response and a thrown `Error`. `runNetworkRegister(name:password:)` looks redundant here.

Reply:
Conditional `catch` can also be eliminated here by implementing `CustomStringConvertible`.